csharp/Spotflow.EventHubConsumer/Program.cs

```diff
@@ -0,0 +1,142 @@
+﻿using Azure.Messaging.EventHubs;
+using Azure.Messaging.EventHubs.Processor;
+using Azure.Storage.Blobs;
+
+using System.Collections.Concurrent;
+using System.Text;
+
+var storageConnectionString = "<< CONNECTION STRING FOR THE STORAGE ACCOUNT >>";
+var blobContainerName = "<< NAME OF THE BLOB CONTAINER >>";
+
+var eventHubsConnectionString = "<< CONNECTION STRING FOR THE EVENT HUBS NAMESPACE >>";
+var eventHubName = "<< NAME OF THE EVENT HUB >>";
+var consumerGroup = "<< NAME OF THE EVENT HUB CONSUMER GROUP >>";
+
+var storageClient = new BlobContainerClient(
+    storageConnectionString,
+    blobContainerName);
+
+var processor = new EventProcessorClient(
+    storageClient,
+    consumerGroup,
+    eventHubsConnectionString,
+    eventHubName);
+
+var partitionEventCount = new ConcurrentDictionary<string, int>();
+
+async Task processEventHandler(ProcessEventArgs args)
+{
+    try
+    {
+        // If the cancellation token is signaled, then the
+        // processor has been asked to stop.  It will invoke
+        // this handler with any events that were in flight;
+        // these will not be lost if not processed.
+        //
+        // It is up to the handler to decide whether to take
+        // action to process the event or to cancel immediately.
+
+        if (args.CancellationToken.IsCancellationRequested)
+        {
+            return;
+        }
+
+        var partition = args.Partition.PartitionId;
+        var eventBody = args.Data.EventBody.ToArray();
+        Console.WriteLine($"Event from partition {partition} with length {eventBody.Length}.");
```

Comment:
Using `Console.Write*` is wrong here and will lead to messed-up console output even for moderate amount of concurrent events (concurrency). It is necessary to use either some locks for writing to Console (very ugly) or `ILogger` (more complex):

```csharp
using var loggerFactory = LoggerFactory.Create(builder => builder.AddConsole());
var logger = loggerFactory.CreateLogger(...);
```

Reply:
Used LoggerFactory